h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -3144,7 +3149,8 @@ public void removeMap(String name) {
         if(id > 0) {
             MVMap<?, ?> map = getMap(id);
             if (map == null) {
-                map = openMap(name, MVStoreTool.getGenericMapBuilder());
+                // Here should use id instead of name because it may be renamed
+                map = openMap(id, MVStoreTool.getGenericMapBuilder());
```

Comment:
why do we care if it was concurrently re-named? End result is an absence of the map with the name specified.

Reply:
Here there are three stories. The first one is removing a new renamed map, the second is removing a new created map, and the last is opening an old name map after it renamed.

Reply:
Attempt to open or remove map (by name), after it was renamed concurrently, should fail, so as removal of the map, which has not been even fully created yet. I wouldn't try to fix it.